docs/general/run-a-node/upgrade-log.md

```diff
@@ -6,6 +6,179 @@ description: >-
 
 # Upgrade Log
 
+## 2022-04-11 (9:00 UTC) - Damask Upgrade
+
+* **Upgrade height:** upgrade is scheduled to happen at epoch **13402**.
+
+:::info
+
+We expect the Mainnet network to reach this epoch at around 2022-04-11 9:00
+UTC.
+
+:::
+
+### Instructions - Voting
+
+:::caution
+
+**Voting for the upgrade proposal will end at epoch 13152. We expect the Mainnet
+network to reach this epoch at around 2022-03-31 22:00 UTC**.
+
+:::
+
+:::info
+
+At this time only entities which have active validator nodes scheduled in the
+validator set are eligible to vote for governance proposals.
+
+:::
+
+:::danger
+
+At least **75%** of the **voting power** needs to cast vote on the upgrade
+proposal for the result to be valid.
+
+At least **90%** of the votes need to be **yes** votes for a proposal to be
+accepted.
+
+:::
+
+The Oasis Protocol Foundation has submitted an [upgrade governance proposal]
+with the following contents:
+
+```yaml
+{
+   "v": 1,
+    "handler": "mainnet-upgrade-2022-04-11",
+    "target": {
+        "consensus_protocol": {
+            "major": 5
+        },
+        "runtime_host_protocol": {
+            "major": 5
+        },
+        "runtime_committee_protocol": {
+            "major": 4
+        }
+    },
+    "epoch": 13402
+}
+```
+
+To view the proposal yourself, you can run the following command on your online
+Oasis Node:
+
+```bash
+oasis-node governance list_proposals -a $ADDR
+```
+
+where `$ADDR` represents the path to the internal Oasis Node UNIX socket
+prefixed with `unix:` (e.g.`unix:/serverdir/node/internal.sock`).
+
+The output should look like:
+
+```yaml
+[
+  {
+    "id": 2,
+    "submitter": "oasis1qpydpeyjrneq20kh2jz2809lew6d9p64yymutlee",
+    "state": "active",
+    "deposit": "10000000000000",
+    "content": {
+      "upgrade": {
+        "v": 1,
+        "handler": "mainnet-upgrade-2022-04-11",
+        "target": {
+          "consensus_protocol": {
+            "major": 5
+          },
+          "runtime_host_protocol": {
+            "major": 5
+          },
+          "runtime_committee_protocol": {
+            "major": 4
+          }
+        },
+        "epoch": 13402
+      }
+    },
+    "created_at": 12984,
+    "closes_at": 13152
+  }
+]
+```
+
+Obtain [your entity's nonce] and store it in the `NONCE` variable. You can do
+that by running:
+
+```yaml
+ENTITY_DIR=<PATH-TO-YOUR-ENTITY>
+ADDRESS=$(oasis-node stake pubkey2address --public_key \
+  $(cat $ENTITY_DIR/entity.json | jq .id -r))
+NONCE=$(oasis-node stake account nonce --stake.account.address $ADDRESS -a $ADDR)
+```
+
+where `<PATH-TO-YOUR-ENTITY>` is the path to your entity's descriptor, e.g.
+`/serverdir/node/entity/`.
+
+To vote for the proposal, use the following command to generate a suitable
+transaction:
+
+```bash
+oasis-node governance gen_cast_vote \
+  "${TX_FLAGS[@]}" \
+  --vote.proposal.id 2 \
+  --vote yes \
+  --transaction.file tx_cast_vote.json \
+  --transaction.nonce $NONCE \
+  --transaction.fee.gas 2000 \
```

Comment:
Is this enough gas? Has someone confirmed?

Reply:
I haven't tried but we had the same instructions for the last on-chain voting: https://docs.oasis.dev/general/run-a-node/upgrade-log#instructions---voting